src/parser/view_transform.c

```diff
@@ -3301,8 +3408,11 @@ mq_copypush_sargable_terms (PARSER_CONTEXT * parser, PT_NODE * statement, PT_NOD
 
   if (statement->node_type == PT_SELECT
       /* never do copy-push optimization for a hierarchical query */
-      && statement->info.query.q.select.connect_by == NULL && spec->info.spec.derived_table_type == PT_IS_SUBQUERY
-      && (derived_table = spec->info.spec.derived_table) && PT_IS_QUERY (derived_table)
+      && statement->info.query.q.select.connect_by == NULL
+      && (spec->info.spec.derived_table_type == PT_IS_SUBQUERY
+	  || spec->info.spec.derived_table_type == PT_DERIVED_DBLINK_TABLE)
+      && (derived_table = spec->info.spec.derived_table)
+      && (PT_IS_QUERY (derived_table) || derived_table->node_type == PT_DBLINK_TABLE)
```

Comment:
How about fixing it like below.
```
if (statement->node_type == PT_SELECT
   && statement->info.query.q.select.connect_by == NULL
   && (derived_table = spec->info.spec.derived_table) 
{
  if ( PT_IS_QUERY (derived_table)
        && (spec->info.spec.derived_table_type == PT_IS_SUBQUERY )
     { 
        mq_copypush_sargable_terms_helper ()
     }
  else if (derived_table->node_type == PT_DBLINK_TABLE
           spec->info.spec.derived_table_type == PT_DERIVED_DBLINK_TABLE)
    {
      mq_copypush_sargable_terms_dblink ()
    }
}
```

Reply:
the below condition should be also added:
  && !PT_SELECT_INFO_IS_FLAGED (statement, PT_SELECT_INFO_IS_MERGE_QUERY at the top as top of the condition.

Reply:
I said it for readability reasons.

Reply:
Ok, good point for readability.